renku_notebooks/api/notebooks.py

```diff
@@ -55,83 +35,95 @@
     FailedParsing,
 )
 from ..util.misc import read_server_options_file
+from .classes.server import UserServer
 
 
 bp = Blueprint("notebooks_blueprint", __name__, url_prefix=config.SERVICE_PREFIX)
 
 
 @bp.route("servers")
-@validate_response_with(
-    {200: {"schema": ServersGetResponse(), "description": "List of all servers."}}
-)
+@marshal_with(ServersGetResponse(), code=200, description="List of all servers")
 @doc(tags=["servers"], summary="Information about all active servers.")
 @authenticated
 def user_servers(user):
     """Return a JSON of running servers for the user."""
-    namespace = request.args.get("namespace")
-    project = request.args.get("project")
-    branch = request.args.get("branch")
-    commit_sha = request.args.get("commit_sha")
-
-    servers = get_user_servers(user, namespace, project, branch, commit_sha)
-    current_app.logger.debug(servers)
-    return jsonify({"servers": servers})
+    user_pods = user.pods
```

Comment:
@ableuler, I already import the `User` class in the same place where I define the `Server` class. If I then import the `Server` class where I define the `User` class so that I can have something like `user.servers` I will get a circular import. That is why I do not have an attribute of the `User` class that returns a list of server objects.

Reply:
What I can do is convert the pods in server classes in the `notebooks.py` script in the line immediately below your comment and then do the filtering on server objects as you suggested. This should work.

Reply:
@ableuler I think it is cleaner to not introduce a 2nd function to filter the user pods - one by k8s pod annotations and another by Server class attributes. And I cannot eliminate the `filter_pods_by_attributes` function because I use it in quite a few places and in some of them I do not have access to a server object. Let me know if you are ok with this.

Reply:
Also the k8s pod is the 'ground truth' for the information about the servers - so I think it makes sense to go straight to the source for filtering, if possible.